pkg/syncer/fullsync.go

```diff
@@ -70,6 +70,13 @@ func CsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer, vc s
 	// k8sPVMap is useful for clean and quicker look up.
 	k8sPVMap := make(map[string]string)
 
+	if migrationFeatureStateForFullSync && len(metadataSyncer.configInfo.Cfg.VirtualCenter) == 1 {
```

Comment:
we do not need this check here.
migration service will get initialized only when `pv.Spec.VsphereVolume != nil ` which means we have in-tree PV on the cluster.
also migration service is no longer a disabled feature. so user is not going to enable it after deployment. 
We can in fact remove all occurrences of

```
			if volumeMigrationService == nil {
				// Instantiate volumeMigrationService when migration feature state is True.
				if err = initVolumeMigrationService(ctx, metadataSyncer); err != nil {
```

Reply:
We don't need  `volumeMigrationService` only when in-tree PVs are present.

See this: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/fullsync.go#L509

Here we are attempting tp delete migration service CR in all cases. If we do not initialise `volumeMigrationService`, it will lead to nil pointer exception.

Reply:
I am now initialising volume migration service as part of metadatasyncer initialisation but only for a single VC setup (as it is not applicable for multi VC deployments).